### PR TITLE
docs: improvements + additions to module docs

### PR DESCRIPTION
**Pull request by shahednasser**
- Split Module and Module Links to their own chapters
- Add new docs on db operations and transactions in modules, multiple services, links with custom columns, etc...
- Added a list of registered dependencies in a module container

### CODE REVIEW

File: www/apps/book/app/advanced-development/modules/db-operations/page.mdx [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+  InjectManager,
+  InjectTransactionManager,
+  MedusaContext
+} from "@medusajs/utils"
+import { Context } from "@medusajs/types"
+import { EntityManager } from "@mikro-orm/knex"
+
+class HelloModuleService {
+  // ...
+  @InjectTransactionManager()
```
Comment (adrien2p) [inline www/apps/book/app/advanced-development/modules/db-operations/page.mdx:317]: here we dont need this decorator since we are creating a transaction in the method body right? is there any example purpose to mix both here, also if there is a need, then we should mention passing the transaction manager of the method to the transaction options in the body

File: www/apps/book/app/advanced-development/modules/db-operations/page.mdx [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+```ts highlights={[["16"]]}
+class HelloModuleService {
+  // ...
+  @InjectTransactionManager()
+  async update_(
+    input: {
+      id: string,
+      name: string
+    },
+    @MedusaContext() sharedContext?: Context<EntityManager>
```
Comment (adrien2p) [inline www/apps/book/app/advanced-development/modules/db-operations/page.mdx:393]:
```suggestion
    @MedusaContext() sharedContext?: Context<SqlEntityManager> // from postgres
```

File: www/apps/book/app/advanced-development/modules/db-operations/page.mdx [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+  ): Promise<any> {
+    return await this.baseRepository_.transaction<EntityManager>(
```
Comment (adrien2p) [inline www/apps/book/app/advanced-development/modules/db-operations/page.mdx:395]: suggestion: same here

File: www/apps/book/app/advanced-development/modules/db-operations/page.mdx [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+      async (transactionManager) => {
+        // ...
+      },
+      {
+        transaction: sharedContext.transactionManager
```
Comment (adrien2p) [inline www/apps/book/app/advanced-development/modules/db-operations/page.mdx:400]: suggestion: by default it is not nested, we need to provide the option

File: www/apps/book/app/advanced-development/modules/db-operations/page.mdx (hunk 1 above)
Comment (shahednasser) [reply www/apps/book/app/advanced-development/modules/db-operations/page.mdx:317]: This was part of our conversation yesterday on whether to always include the decorator. Then the fix here would be to pass the transaction to the `transaction` method, right?

File: www/apps/book/app/advanced-development/modules/db-operations/page.mdx (hunk 2 above)
Comment (shahednasser) [reply www/apps/book/app/advanced-development/modules/db-operations/page.mdx:393]: Actually I checked and `SqlEntityManager` from `@mikro-orm/knex` is the same as importing `EntityManager` from `@mikro-orm/knex`, but maybe it's better to include the import in all code snippets to avoid confusion.

File: www/apps/book/app/advanced-development/modules/isolation/page.mdx [hunk 5]
```diff
@@ -21,12 +21,91 @@ For example, your custom module can't resolve the Product Module's main service
...
+  async (_, { container }) => {
+    const brandModuleService = container.resolve(
+      "brandModuleService"
+    )
+
+    const brands = await brandModuleService.listBrands()
+
+    return new StepResponse(brands)
+  }
+)
```
Comment (adrien2p) [inline www/apps/book/app/advanced-development/modules/isolation/page.mdx:73]: nit(non blocking): should we show the useRemoteQueryStep?

File: www/apps/book/app/advanced-development/modules/isolation/page.mdx [hunk 6]
```diff
@@ -21,12 +21,91 @@ For example, your custom module can't resolve the Product Module's main service
...
+const createBrandsInCmsStep = createStep(
+  "create-brands-in-cms",
+  async ({ brands }, { container }) => {
+    const cmsModuleService = container.resolve(
+      "cmsModuleService"
+    )
+
+    const cmsBrands = await cmsModuleService.createBrands(brands)
+
+    return new StepResponse(cmsBrands, cmsBrands)
```
Comment (adrien2p) [inline www/apps/book/app/advanced-development/modules/isolation/page.mdx:84]: suggestion: it is not necessary to pass both args if they are the same, by default the compensation input will be the first arg if the second is not provided

File: www/apps/book/app/advanced-development/modules/isolation/page.mdx [hunk 7]
```diff
@@ -21,12 +21,91 @@ For example, your custom module can't resolve the Product Module's main service
...
+  },
+  async (brands, { container }) => {
+    const cmsModuleService = container.resolve(
+      "cmsModuleService"
+    )
+
+    await cmsModuleService.deleteBrands(brands)
```
Comment (adrien2p) [inline www/apps/book/app/advanced-development/modules/isolation/page.mdx]: Q: wouldnt it be brandIds?

**Review by adrien2p — APPROVED** [review]

File: www/apps/book/app/advanced-development/module-links/custom-columns/page.mdx [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+
+export const linkHighlights = [
+  ["10", "extraColumns", "Custom columns to add to the created link's table."],
+  ["11", "metadata", "The column's name."],
+  ["12", "type", "The column's type."]
+]
+
+```ts highlights={linkHighlights}
+import HelloModule from "../modules/hello"
+import ProductModule from "@medusajs/product"
```
Comment (adrien2p) [inline www/apps/book/app/advanced-development/module-links/custom-columns/page.mdx]: Q: will all import be changed in another pr?

Comment (shahednasser) [reply www/apps/book/app/advanced-development/module-links/custom-columns/page.mdx]: No I'll be updating this PR in a bit

File: www/apps/book/app/advanced-development/modules/isolation/page.mdx (hunk 5 above)
Comment (shahednasser) [reply www/apps/book/app/advanced-development/modules/isolation/page.mdx:73]: In this case I want to showcase the usage of the different modules, so I think it's fine to not use `useRemoteQueryStep`.

File: www/apps/book/app/advanced-development/modules/isolation/page.mdx (hunk 6 above)
Comment (shahednasser) [reply www/apps/book/app/advanced-development/modules/isolation/page.mdx:84]: I know, but I've avoided documenting this to avoid confusion.

File: www/apps/book/app/advanced-development/modules/isolation/page.mdx (hunk 7 above)
Comment (shahednasser) [reply www/apps/book/app/advanced-development/modules/isolation/page.mdx]: fixed 👍🏻

**Comment by shahednasser** [comment]
@olivermrbl can we get this one in?

**Review by olivermrbl — APPROVED** [review]
